hack/deploy.sh

```diff
@@ -0,0 +1,274 @@
+#!/bin/bash
+
+# Copyright 2018 Oracle and/or its affiliates. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Utility functions for managing the version of the CCM running in a Kubernetes
+# cluster.
+# 
+# Functions for deploying the specified version of the CCM in a K8s cluster that 
+# that already has a DaemonSet deployed CCM. The version should be defined by the  
+# environment variable VERSION.
+# 
+# Functions for managing a soft (no-threadsafe) lock on the CCM daemonset to 
+# minimise it being upgraded/downgraded during integration tests.
+#
+# Kubectl configured to point to the required target cluster is required on the 
+# host machine.
+
+
+# The root name of the CCM daemon-set and pods.
+CCM_NAME="oci-cloud-controller-manager"
+# The name of an annotation that can be used to 'lock' the CCM from upgrade/
+# downgrade operations. 
+CCM_LOCK_LABEL="ccm-deployment-lock"
```

Comment:
We add a simple kube annotation to the CCM daemon-set to help manage upgrade/downgrade operations.